src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -448,20 +448,7 @@ private Object HandleToObject(IntPtr handle)
 
         private bool Get_CORINFO_METHOD_INFO(MethodDesc method, MethodIL methodIL, CORINFO_METHOD_INFO* methodInfo)
         {
```

Comment:
Both crossgen1 and JIT are bailing out for PInvokes from getMethodInfo as well here: https://github.com/dotnet/runtime/blob/891e8d4d4f966d8888c59f38a4ad9f8c2878adf6/src/coreclr/vm/jitinterface.cpp#L7783 

Why is this change necessary?

Reply:
This change does not appear to be fixing the x86 crash based on my quick local test. Were you able to confirm locally that it is fixing the problem?

Reply:
While I believe I see a CG1 vs. CG2 diff in NgenDump I tracked down to this difference, I have rolled back the 2nd commit based on your clarification that the error I've been attempting to fix is unrelated to my change. Amusingly this is about the 2nd or 3rd time the SPC CG2 switchover revealed a bug that ended up being completely unrelated to the switchover as such. Let's hope it just means that we do take CG2 testing seriously ;-).

Reply:
> While I believe I see a CG1 vs. CG2 diff in NgenDump I tracked down to this difference,

Do you still see the diff in the current master? If yes, could you please share the details of the diff?